nvflare/client/api_spec.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Optional
+
+from nvflare.apis.analytix import AnalyticsDataType
+from nvflare.app_common.abstract.fl_model import FLModel
+
+CLIENT_API_KEY = "CLIENT_API"
+CLIENT_API_TYPE_KEY = "CLIENT_API_TYPE"
+
+
+class APISpec(ABC):
+    @abstractmethod
+    def init(self, rank: Optional[str] = None):
```

Comment:
need to add docstrings to these methods in spec,
you can copy some from https://github.com/NVIDIA/NVFlare/blob/main/nvflare/client/api.py